pipe.go

```diff
@@ -89,6 +91,10 @@ type pipe struct {
 	recvs           int32
 	r2ps            bool // identify this pipe is used for resp2 pubsub or not
 	noNoDelay       bool
+	lftm            time.Duration // lifetime
+	lftmMu          sync.Mutex    // guards lifetime timer
```

Comment:
Do we really need the mutex and the bool flag?

Reply:
I reviewed again, we don't need bool flag.
I thought that time.Reset and time.Stop need mutex when using `<= go 1.22 `. Maybe I've got it wrong.

Reply:
The source looks like it is thread-safe https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/runtime/time.go;l=314.

Reply:
Thanks, As you said, it looks like thread-safe. I will remove it.

I misread `This cannot be done concurrent to other receives from the Timer's channel or other calls to the Timer's Stop method.` of https://pkg.go.dev/time@go1.21.13#Timer.Stop . Sorry.

Reply:
And we are using the `AfterFunc` timer which has no channel associated.

Reply:
That's right.

Reply:
Fixed [f950c1e](https://github.com/redis/rueidis/pull/727/commits/f950c1e7294962a119774ffb33f4b6271278d932)